app/src/main/kotlin/com/flxrs/dankchat/DankChatViewModel.kt

```diff
@@ -1,31 +1,61 @@
 package com.flxrs.dankchat
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.flxrs.dankchat.data.ChatRepository
 import com.flxrs.dankchat.data.DataRepository
+import com.flxrs.dankchat.data.api.ApiManager
+import com.flxrs.dankchat.preferences.DankChatPreferenceStore
+import com.flxrs.dankchat.utils.extensions.withoutOAuthSuffix
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class DankChatViewModel @Inject constructor(
     private val chatRepository: ChatRepository,
+    private val dankChatPreferenceStore: DankChatPreferenceStore,
+    private val apiManager: ApiManager,
     dataRepository: DataRepository,
 ) : ViewModel() {
 
     val commands = dataRepository.commands
     var started = false
         private set
 
+    private val _oauthResult = MutableSharedFlow<OAuthResult>(extraBufferCapacity = 2, onBufferOverflow = BufferOverflow.DROP_OLDEST)
```

Comment:
I still think we should use a channel here, as a SharedFlow has the downside of dropping elements if there is no subscriber. So in the rare edge-case where the user rotates his phone while we send our result we would not receive the result:
```kotlin
private val _oauthResult = Channel<OAuthResult>(Channel.BUFFERED)
val oAuthResult = _oauthResult.receiveAsFlow()
```
Also, can we rename the sealed class and properties here to maybe `ValidationResult`?

Reply:
OK, I'll rename as you suggested below :+1:

Reply:
Change it to a `Channel` instead of `SharedFlow` and we're good to go